File: site/testsuites/testsuite-javascript/__tests__/tbdex/pfi/offeringsApiProvider.js
import { MockOfferingsApiProvider } from '../../utils/mockOfferingsApiProvider'
import { Parser } from '@tbdex/http-server'

export class OfferingsApiProvider extends MockOfferingsApiProvider {

    constructor(pfiDid) {
        super();
        this.pfiDid = pfiDid;
    }
  
    // :snippet-start: pfiOverviewReadOfferingsJs
    async getOffering(id) {
        return this.dataProvider.get('offering', id).then(async (result) => {
            if(result){
                return await Parser.parseResource(result);
            }
        });
    }
    
  
    async getOfferings() {
        return this.dataProvider.query('offering', "*").then((results) => {
            const offerings = [];
            
            for (let result of results) {
                const offering = Parser.rawToResourceModel(result);
                offerings.push(offering);
            }

            return offerings;
        });
    }
    
    async setOffering(offering) {
        await this.dataProvider.insert('offering', {
            offeringid: offering.metadata.id,
            payoutcurrency: offering.data.payout.currencyCode,
            payincurrency: offering.data.payin.currencyCode,
            offering: Parser.rawToResourceModel(offering)
        });
    }
    // :snippet-end:
}

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/pfi/OfferingsApiProvider.kt
package website.tbd.developer.site.docs.tbdex.pfi

import tbdex.sdk.protocol.models.*
import website.tbd.developer.site.docs.utils.*

class OfferingsApiProvider: MockOfferingsApiProvider() {

    // :snippet-start: pfiOverviewReadOfferingsKt
    override fun getOffering(id: String): Offering {
        val result = dataProvider.get("offering", id) as String
        return Offering.parse(result)
    }

    override fun getOfferings(): List<Offering> {
        val results = dataProvider.query("offering", "*")
        val offerings = mutableListOf<Offering>()

        for (result in results) {
            val offering = Offering.parse(result as String)
            offerings.add(offering)
        }

        return offerings
    }

    fun setOffering(offering: Offering) {
        val resource = mapOf(
            "offeringid" to offering.metadata.id,
            "payoutcurrency" to offering.data.payout.currencyCode,
            "payincurrency" to offering.data.payin.currencyCode,
            "offering" to offering.toString()
        )
        dataProvider.insert("offering", resource)
    }
  // :snippet-end:
}
